core/src/trezor/workflow.py

```diff
@@ -178,16 +192,24 @@ async def _timeout_task(self, callback: IdleCallback) -> None:
         self.tasks[callback] = self._timeout_task(callback)
         callback()
 
-    def touch(self) -> None:
+    def touch(self, keep_deadline: bool = False) -> None:
         """Wake up the idle timer.
 
         Events that represent some form of activity (USB messages, touches, etc.) should
         call `touch()` to notify the timer of the activity. All pending callback timers
         will reset.
         """
+        if not dont_autolock:
```

Comment:
this seems wrong? means that in the weird case where a dont_autolock workflow is waiting for user confirmation, it will remain unlocked.

Reply:
Isn't that what we want though, for the whitelisted workflows, to remain unlocked? With locked storage.

I've used the the same whitelist for not waking up the timer because I'm not sure whether we can entirely ignore usb messages. That would break cases like `test_lots_of_inputs` with short autolock delay, which currently work fine?

Reply:
Ok, let's handle it in apps/bitcoin ...

Reply:
> That would break cases like test_lots_of_inputs with short autolock delay, which currently work fine?

right, which is why I wanted bitcoin signing to set its own exception to autolock too.

ignoring USB messages seems overall more correct, if we have a mechanism to let the workflow finish